content/docs/reference-glossary.md

```diff
@@ -7,38 +7,58 @@ permalink: docs/glossary.html
 
 ## Bundlers
-Bundlers put all of your JavaScript code & dependency into one "bundle", usually into one file. Some bundlers commonly used in React applications include: Webpack and Browserify.
+
+Bundlers take JavaScript and CSS code written as separate modules (often hundreds of them), and combine them together into a few files better optimized for the browsers. Some bundlers commonly used in React applications include [Webpack](https://webpack.js.org/) and [Browserify](http://browserify.org/).
 
 ## Package Manager 
-Package managers are tools that allow you to manage dependencies in your project. npm & Yarn are two package managers commonly used in React applications.
+
+Package managers are tools that allow you to manage dependencies in your project. [npm](https://www.npmjs.com/) and [Yarn](http://yarnpkg.com/) are two package managers commonly used in React applications. Both of them are clients for the same npm package registry.
 
 ## CDN
+
 CDN stands for Content Delivery Network. CDNs deliver cached, static content from a network of servers across the globe. 
 
 ## JSX
-JSX is a syntax extension to JavaScript. It is similar to a template language. JSX gets compiled to `React.createElement()` calls which return plain JavaScript objects called 'React elements'. To get a basic introduction to JSX [see the docs here](/docs/introducing-jsx.html) and find a more in-depth tutorial on JSX [here](/docs/jsx-in-depth.html)
 
-React DOM uses camelCase property naming convention instead of HTML attribute names. For example, tabindex becomes tabIndex in JSX. The attribute class is also written as className since `class` is a reserved word in JavaScript.
+JSX is a syntax extension to JavaScript. It is similar to a template language, but it has full power of JavaScript. JSX gets compiled to `React.createElement()` calls which return plain JavaScript objects called "React elements". To get a basic introduction to JSX [see the docs here](/docs/introducing-jsx.html) and find a more in-depth tutorial on JSX [here](/docs/jsx-in-depth.html).
+
+React DOM uses camelCase property naming convention instead of HTML attribute names. For example, `tabindex` becomes `tabIndex` in JSX. The attribute `class` is also written as `className` since `class` is a reserved word in JavaScript:
+
+```js
+const name = 'Clementine';
+ReactDOM.render(
+  <h1 className="hello">My name is {name}!</h1>,
+  document.getElementById('root')
+);
+```  
 
 ## [Elements](/docs/rendering-elements.html)
+
 React elements are the building blocks of React applications. One might confuse elements with a more widely known concept of "components". Elements are what components are "made of". An element describes what you want to see on the screen. React elements are immutable.
```

Comment:
I still think the wording here is a bit vague:
> Elements are what components are "made of".

I wonder if we could improve it by changing it to something like:
> Elements are a template for creating components.

Reply:
I just removed this sentence since it's duplicating the same thought as in first sentence and then after the snippet.